src/index/updater.rs

```diff
@@ -130,7 +132,7 @@ impl Updater {
         self.commit(wtx, value_cache)?;
```

Comment:
I think there should be a commit test on line 131 here. Something like:

```
let shouldCommitForSavepoint = starting_height - self.height < 15 && (self.height + uncommitted) % 5 === 0;

if uncommitted == 5000 || shouldCommitForSavepoint {
```

Disclaimer: Terrible variable naming and the logic might be off by 1 😋